flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2711,6 +2710,24 @@ def test_create_fails_on_mounted_filesystem(self):
             scenario.deployer))
         self.assertEqual(afile.getContent(), b"data")
 
+    def test_create_fails_on_existing_filesystem(self):
+        """
+        Running ``CreateFilesystem`` on a block device that already has a file
+        system fails with an exception and preserves the data.
+
+        This is because mkfs is a destructive operation that will destroy any
+        existing filesystem on that block device.
+        """
+        mountroot = mountroot_for_test(self)
+        mountpoint = mountroot.child(b"mount-test")
+        scenario = self._run_success_test(mountpoint)
+        afile = mountpoint.child(b"file")
+        afile.setContent(b"data")
+        # Unmount the filesystem
+        run_process([b"umount", mountpoint.path])
+        # Try recreating filesystem; this should fail.
+        self.failureResultOf(scenario.create(), FilesystemExists)
```

Comment:
Oh, I did that, cut it temporarily and forgot to yank. Added. 

Refactoring done. Probably could do more if we had a declaratively way of making files, ala [treeshape](https://github.com/jml/treeshape).